Sources/UIScreen.swift

```diff
@@ -6,29 +6,300 @@
+
+        GPU_BlitTransform(
+            image.rawPointer,
+            nil,
+            self.rawPointer,
+            Float(contentsGravity.offset.x),
+            Float(contentsGravity.offset.y),
+            0, // rotation in degrees
+            Float(contentsGravity.scale.width / contentsScale),
+            Float(contentsGravity.scale.height / contentsScale)
+        )
+    }
+
+    func setShapeBlending(_ newValue: Bool) {
+        GPU_SetShapeBlending(newValue)
+    }
+
+    func setShapeBlendMode(_ newValue: GPU_BlendPresetEnum) {
+        GPU_SetShapeBlendMode(newValue)
+    }
+
+    func clear() {
+        GPU_Clear(rawPointer)
+    }
+
+    var clippingRect: CGRect? {
+        didSet {
+            guard let clippingRect = clippingRect else {
+                return GPU_UnsetClip(rawPointer)
+            }
+
+            GPU_SetClipRect(rawPointer, clippingRect.gpuRect(scale: scale))
+        }
+    }
+
+    func fill(_ rect: CGRect, with color: UIColor, cornerRadius: CGFloat) {
+        if cornerRadius >= 1 {
+            GPU_RectangleRoundFilled(rawPointer, rect.gpuRect(scale: scale), cornerRadius: Float(cornerRadius), color: color.sdlColor)
+        } else {
+            GPU_RectangleFilled(rawPointer, rect.gpuRect(scale: scale), color: color.sdlColor)
+        }
+    }
+
+    func outline(_ rect: CGRect, lineColor: UIColor, lineThickness: CGFloat) {
+        GPU_SetLineThickness(Float(lineThickness))
+        GPU_Rectangle(rawPointer, rect.gpuRect(scale: scale), color: lineColor.sdlColor)
+    }
+
+    func outline(_ rect: CGRect, lineColor: UIColor, lineThickness: CGFloat, cornerRadius: CGFloat) {
+        if cornerRadius > 1 {
+            GPU_SetLineThickness(Float(lineThickness))
+            GPU_RectangleRound(rawPointer, rect.gpuRect(scale: scale), cornerRadius: Float(cornerRadius), color: lineColor.sdlColor)
+        } else {
+            outline(rect, lineColor: lineColor, lineThickness: lineThickness)
+        }
+    }
+
+    func flip() throws {
+        GPU_Flip(rawPointer)
+        try throwOnErrors(ofType: [GPU_ERROR_USER_ERROR, GPU_ERROR_BACKEND_ERROR])
+    }
+
+
+    func destroy() {
+        UIView.layersWithAnimations.removeAll()
+        UIEvent.activeEvents.removeAll()
+        UIView.currentAnimationPrototype = nil
+        UIFont.clearCaches()
+
+        if rawPointer == nil { return } // dummy screen or already destroyed
+        defer { GPU_Quit() }
+
+        // get and destroy existing GLRenderer because only one SDL_Window can exist on Android at the same time
+        guard let gpuContext = self.rawPointer.pointee.context else {
+            assertionFailure("glRenderer gpuContext not found")
+            return
+        }
+
+        let existingWindowID = gpuContext.pointee.windowID
+        let existingWindow = SDL_GetWindowFromID(existingWindowID)
+        SDL_DestroyWindow(existingWindow)
+        rawPointer = nil
+        print("Destroyed UIScreen")
+    }
+
+    deinit {
+        destroy()
```

Comment:
do we use `destroy` in any other places any more? this was initially a workaround to allow us to deinit the Screen before the instance gets deinitted, but it may make sense to move the body of `destroy` into `deinit`